### DISCUSSION TITLE
After updating cadence server to newest version it stops working

### DISCUSSION THREAD

**Discussion by rafalmnich** [uber/cadence#4578] [Q&A]
I have this error when trying to run any workflow:

Error while dialing dial tcp: address [:7835: missing ']' in address.

Looks like a configuration issue, but I cannot find it anywhere.

**Comment by longquanzheng** [comment]
Hi @rafalmnich , can you add more details? 
Did you upgrade to 0.22.x? 

Looks like it's trying to connect to the gRPC port: https://github.com/uber/cadence/blob/master/config/development.yaml#L40 

It would be nice if you can provide more detailed logs and how you configured you server. 

Thanks

**Reply by rafalmnich** [reply]
Yes, I was updating to 0.22.x. And before we had 0.18.x. The problem was in configuration, overwriting `BIND_ON_IP`. After removing this overwrite, it started to work. But I still have some problem with resolving DNS. Where do I setup the DNS host / IP?
I get the `Failed to update DNS` error with `lookup cadence on 10.1.0.2:53: no such host`.

**Reply by longquanzheng** [reply]
Can you show your config? Set up DNS host/IP for which config? I know one place that you need to set DNS host port is https://github.com/uber/cadence/blob/master/config/development.yaml#L83 

You need to put frontend service DNS:port there.

**Reply by rafalmnich** [reply]
I am using your docker container, with mostly default values and I have it deployed in ecs in aws. I use service discovery with a dns server. The error showing a problem with connecting to port `53` and the line you send me in the config has `7933` as default port.

**Comment by longquanzheng** [comment]
You might also want to read this doc about gRPC feature 
cc @vytautas-karpavicius
